### PR TITLE
Use `-D warnings` instead of `deny-warnings` feature.

### PR DESCRIPTION
**Pull request by Jarcho**
r? @flip1995 
changelog: none

### CODE REVIEW

File: .github/workflows/clippy.yml [hunk 1]
```diff
@@ -47,25 +47,25 @@ jobs:
 
     # Run
     - name: Build
-      run: cargo build --tests --features deny-warnings,internal
+      run: cargo build --tests --features internal -D warnings
```
Comment (flip1995) [inline .github/workflows/clippy.yml:50]: Clippy is the only `cargo` command that takes `-D` flags. We have documentation in our book how to do this in GHA: https://doc.rust-lang.org/clippy/continuous_integration/github_actions.html

TL;DR: Add this near the top of the workflow file

```yaml
env:
  RUSTFLAGS: "-Dwarnings"
```

Comment (Jarcho) [reply .github/workflows/clippy.yml]: That would be better even if the other way did work.

**Comment by flip1995** [comment]
```
failures:
    test_no_deps_ignores_path_deps_in_workspaces
```

Oh, this is a good catch. `path_dep` should be changed to trigger a Clippy lint, not a rustc lint.

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**